collect_app/src/main/java/org/odk/collect/android/formentry/FormEndView.kt

```diff
@@ -19,15 +20,18 @@ class FormEndView(
     init {
         binding.description.text = context.getString(R.string.save_enter_data_description, formTitle)
 
-        binding.saveAsDraft.isEnabled = isSaveAsDraftEnabled
+        binding.saveAsDraft.isEnabled = formEndViewModel.isSaveDraftEnabled()
         binding.saveAsDraft.setOnClickListener {
             listener.onSaveClicked(false)
         }
 
-        binding.finalize.isEnabled = isFinalizeEnabled
+        binding.finalize.isEnabled = formEndViewModel.isFinalizeEnabled()
         binding.finalize.setOnClickListener {
             listener.onSaveClicked(true)
         }
+        if (formEndViewModel.shouldFormBeSentAutomatically(form)) {
```

Comment:
I think it would be good for `FormEndViewModel` to encapsulate any accesses to `Form` here so that we don't need to pass one in to `FormEndView` for this. The ViewModel could use a similar method to `FormEntryViewModel`, `FormSaveViewModel` etc to load the form from the `FormSessionRepository`.

I think the `formTitle` param should really be dropped and should also be something the view gets from the ViewModel. This might require quite a chunk of rework however (as this data gets passed around between ViewModels), so I could understand it feeling like something we should punt on for the moment.

Reply:
I agree regarding the first part but I wasn't sure how to handle this to make the code clean. What do you think about https://github.com/getodk/collect/pull/5569/commits/477b025183961b40d9b1443487b7a03caff1a90a and keeping a `formController` and a `form` objects in the same repository? I think creating a separate repository just for the `form` object would be weird, they are both kind of a `formSession` so probably we should keep the both in the same place.

Reply:
That makes a lot of sense to me! I could see that we'll probably also want `Instance` (probably optional) on the form session eventually.